go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -1450,6 +1451,32 @@ func (nc *DefaultNodeNetworkController) reconcileConntrackUponEndpointSliceEvent
 					netlink.ConntrackReplyAnyIP, nil); err != nil {
 					klog.Errorf("Failed to delete conntrack entry for %s: %v", oldIPStr, err)
 				}
+
+				// Delete conntrack entries for NodePort when the node is on a remote node, using nodePort for --orig-port-src match
+				// For example: udp      17 116 src=172.18.0.2 dst=172.18.0.4 sport=35254 dport=31832 src=10.244.0.8 dst=172.18.0.2 sport=8080 dport=35254 [ASSURED] mark=10 secctx=system_u:object_r:unlabeled_t:s0 zone=16 use=1
+				// TODO: Once vishvananda/netlink support ConntrackFilterType '--reply-port-src', we can use oldPort.Port directly.
+				if svc.Spec.Type == corev1.ServiceTypeNodePort {
```

Comment:
was this an issue that always existed?
I remember fixing this during the olden days for UDN services https://github.com/ovn-kubernetes/ovn-kubernetes/pull/2829/files
was it that we didn't do it for nodeports? OR did we regress somewhere?

Reply:
> was this an issue that always existed?

I think so. I don't think this is a regression. The existing code doesn't consider the nodePort service case.